catroid/src/org/catrobat/catroid/formulaeditor/SensorHandler.java

```diff
@@ -117,50 +155,90 @@ public static Double getSensorValue(Sensors sensor) {
 		switch (sensor) {
 
 			case X_ACCELERATION:
-				return (double) instance.linearAcceleartionX;
+				return (double) instance.linearAccelerationX;
 
 			case Y_ACCELERATION:
-				return (double) instance.linearAcceleartionY;
+				return (double) instance.linearAccelerationY;
 
 			case Z_ACCELERATION:
-				return (double) instance.linearAcceleartionZ;
+				return (double) instance.linearAccelerationZ;
 
 			case COMPASS_DIRECTION:
 				float[] orientations = new float[3];
-				android.hardware.SensorManager.getRotationMatrixFromVector(instance.rotationMatrix,
-						instance.rotationVector);
+				if (!instance.useRotationVectorFallback) {
+					android.hardware.SensorManager.getRotationMatrixFromVector(instance.rotationMatrix,
+							instance.rotationVector);
+				}
 				android.hardware.SensorManager.getOrientation(instance.rotationMatrix, orientations);
 				sensorValue = (double) orientations[0];
 				return sensorValue * RADIAN_TO_DEGREE_CONST * -1f;
 
 			case X_INCLINATION:
-
-				orientations = new float[3];
-				android.hardware.SensorManager.getRotationMatrixFromVector(instance.rotationMatrix,
-						instance.rotationVector);
-				android.hardware.SensorManager.getOrientation(instance.rotationMatrix, orientations);
-				sensorValue = (double) orientations[2];
-				return sensorValue * RADIAN_TO_DEGREE_CONST * -1f;
-
+				if (instance.useRotationVectorFallback) {
+					float rawInclinationX = RADIAN_TO_DEGREE_CONST * (float) (Math.acos(instance.accelerationXYZ[0]));
+					float correctedInclinationX = 0;
+
+					if (rawInclinationX >= 90 && rawInclinationX <= 180) {
+						if (instance.signAccelerationZ > 0) {
+							correctedInclinationX = -(rawInclinationX - 90);
+						} else {
+							correctedInclinationX = -(180 + (90 - rawInclinationX));
+						}
+					} else
+						if (rawInclinationX >= 0 && rawInclinationX < 90) {
```

Comment:
That's an interesting case of else-if coding style...
Do you read it as ```else { if (...``` (because a else block doesn't require braces) or as ```else if (...```?
However, let's see what the android studio formatter does! :wink:

Also removing comments, as suggested by @thmq, does not always end up by just removing them. Please take a look at Clean Code, "Don't use a comment when you can use a function or variable".

However, I prefer this example
```java
// Bad:
// Check to see if the employee is eligible for full benefits
if ((employee.flags & HOURLY_FLAG) && (employee.age > 65))

// Good:
if (employee.isEligibleForFullBenefits())
```